compiler/rustc_resolve/src/lib.rs

```diff
@@ -2638,6 +2639,8 @@ impl<'a> Resolver<'a> {
                                                 is_type: true,
                                             },
                                         );
+                                    } else {
+                                        self.session.delay_span_bug(span, CG_BUG_STR);
```

Comment:
You don't need this in an `else`, `delay_span_bug` will early exit if an error has already been emitted.
(same for the other case)

Reply:
i personally prefer delay span in an else branch if the other branch *obviously* emits an error.

That's just a stylistic preference though. Changed it.